encrypt.go

```diff
@@ -35,9 +35,49 @@ func NewEncryptManager(passphrase string) *EncryptManager {
 	return &EncryptManager{[]byte(passphrase)}
 }
 
-// EncryptGCM encrypts given io.Reader using AES256-GCM
+// Encrypt is used to handle encryption of objects, specifying the desired encryption method to be used
+// It returns a map, whose data depends on the encryption method being used
+//
+// For AES256-CFB, it is as follows:
+//		"encryptedData" -> byte slice containing the encrypted data
+//
+// for AES256-GCM, it is as follows:
+//		"encryptedData" -> byte slice containing the encrypted data
+//		"decryptData" -> is an encrypted, and formatted byte slice containing the cipher and nonceencMethod
+//
+// AES256-GCM mode uses the passphrase supplied when instantiating EncryptManager to
+// encrypt the randomly generated cipherKey, and nonce
+func (e *EncryptManager) Encrypt(r io.Reader, encryptionMethod string) (map[string][]byte, error) {
+	out := make(map[string][]byte)
+	switch strings.ToLower(encryptionMethod) {
+	case "aes256-gcm":
+		encryptedData, nonce, cipherKey, err := e.encryptGCM(r)
+		if err != nil {
+			return nil, err
+		}
+		decryptData, err := e.encryptCipherAndNonce(cipherKey, nonce)
+		if err != nil {
+			return nil, err
+		}
+		out = map[string][]byte{
+			"encryptedData": encryptedData,
+			"decryptData":   decryptData,
```

Comment:
what's this?

Reply:
`map[string][]byte` seems like a very poor output usability and consistency wise - we should retain the old output (`([]byte, error)`)

Reply:
Very true